src/v2/System/useFeatureFlag.tsx

```diff
@@ -1,18 +1,39 @@
 import { useSystemContext } from "v2/System"
+import { Variant } from "unleash-client"
 
-export function useFeatureFlag(flagName: string): boolean | null {
+export type FeatureFlags = Record<string, FeatureFlagDetails>
+
+interface FeatureFlagDetails {
+  flagEnabled: boolean
+  variant: Variant
+}
+
+export function useFeatureFlag(featureName: string): boolean | null {
   const { featureFlags } = useSystemContext()
-  if (featureFlags === undefined) {
-    console.error("[Force] Error: featureFlags is undefined in SystemContext")
+  const flagEnabled = featureFlags?.[featureName]?.flagEnabled
+
+  if (flagEnabled === undefined) {
+    console.error(
+      "[Force] Error: cannot find flagName in featureFlags: ",
+      featureFlags
+    )
     return null
   }
 
-  if (!flagName) {
-    console.error("[Force] Error: no argument passed into useFeatureFlag")
+  return flagEnabled
+}
+
+export function useFeatureVariant(featureName: string): Variant | null {
+  const { featureFlags } = useSystemContext()
+  const variant = featureFlags?.[featureName]?.variant
+
+  if (variant === undefined) {
```

Comment:
ditto - here -- `!variant` is a wider match and more commonly seen around code

Reply:
Ah, for sure. Makes sense here. I group change the logic where I didn't need to.